### DISCUSSION TITLE
怎么指定target只编译，不执行默认的安装？

### DISCUSSION THREAD

**Discussion by bamlan** [xmake-io/xmake#2288] [Q&A] [answered]
有些shared类型的target是不需要安装到系统目录里的，但在build时需要生成，但是set_default(false) 默认情况下直接就不编译了，只能手动编译，有什么api能使target禁用默认的install行为吗？

**Comment by xq114** [comment, marked as answer]
on_install(function(target) end)

**Reply by bamlan** [reply]
OK,　在文档里找到了，谢谢！
[target:on_install](https://xmake.io/#/zh-cn/manual/project_target?id=targeton_install)
[自定义安装脚本](https://xmake.io/#/zh-cn/manual/project_target?id=%e8%87%aa%e5%ae%9a%e4%b9%89%e5%ae%89%e8%a3%85%e8%84%9a%e6%9c%ac)
覆盖target目标的xmake [i|install}的安装操作，实现自定义安装过程。

例如，将生成的apk包，进行安装。

target("test")

    -- 设置自定义安装脚本，自动安装apk文件
    on_install(function (target)

        -- 使用adb安装打包生成的apk文件
        os.run("adb install -r ./bin/Demo-debug.apk")
    end)
